locomotor/include/locomotor/locomotor_action_server.h

```diff
@@ -51,12 +52,16 @@ class LocomotorActionServer
   void publishFeedback(const locomotor_msgs::NavigationState& nav_state);
   void completeNavigation();
   void failNavigation(const locomotor_msgs::ResultCode& result_code);
+
+  // Note: This is a separate method so as not to change the previous constructor parameter list
+  void setPremptionCallback(PreemptionCallback cb) { preemption_cb_ = cb; }
```

Comment:
I'd recommend just having a `virtual void onPreemption() {}` method that the user can override.

Reply:
In which class?

Reply:
In this one. Rather than setting a callback functor, you can just let the user override the `onPreemption` method if they want to trigger some callback. These 2 actions will have equal output.